### ISSUE TITLE
Images, production and development repository question

### ISSUE THREAD

**Issue by be-codified** [dockware/dockware#25]
Hi

This is more of a question, not an issue.

I am using **DEV image** as I will be developing more or less plugins for my shop. I see that DEV image is using **Shopware production repository**.

I am reading Shopware production repository readme and they are saying "If you want to contribute to the Shopware Platform or **develop store plugins**, you should use the **development template**." which is right opposite what you are suggesting - use DEV image (having production template) if you want to develop plugins.

Can you please give me some feedback on this matter?

I also noticed that if I am using production repository, I get "Cannot find declaration" in PHP file when trying to go to specific class, which is not the case with development template. Probably class definitions would come in handy while developing plugins (autocomplete, hinting etc.).

So, at the end, should I use Contribute image (having development template)?

Thank you for your answer.

**Comment by boxblinkracer** [comment]
hi

the reason why shopware writes this, is because it comes with all kinds of dev stuff. they somehow mix the purpose of psh with things for contribution and with things for stable release-dev-stuff

its more like this...what would you use for your production shop? any (probably untagged) commit of the github master, or a real release.
you can use both of course, but for us the correct and reliable answer would be to use a github version for contribution (or very rare fallbacks) and to use the officially released version based on the ZIP download from the website without any stuff you dont need when not contributing (prod template) for our plugins or client shops.
its like using a beta-dev version or stable release without internal dev stuff....

the dockware play is just the production template. 
the dockware dev is production template + dev stuff.
so we use the dev version (prod + dev tools) for all our projects, and that should actually work.

i haven't heard of a "declaration" problem. so this sounds more like a code related problem?
in dockware dev, theres also a sample plugin that works...can you reproduce that issue in there?
then i can try to help you :)

also theres a new page online with that stuff, it might help you too
https://dockware.io/getstarted

cheers :)

**Comment by be-codified** [comment]
Hi,

thanks for the feedback :)

About "declarations", I was able to solve it. Let me rephrase my question: I was asking about PHP intellisense (i.e. help by going to class declaration, autocomplete code like class methods etc.)

This has been solved by putting core files into my local repo and ignore to track them by Git. This way IDE or code editor (in my case VSC) with some kind of PHP intellisense plugin indexes files and you are good to go.

I hope it is clear now. I will close this issue down.

**Comment by boxblinkracer** [comment]
aaaa ok :D
yes, then you need the code locally
if you do not use the bind-mount option, thats the reason why to download the source code
if you use bind-mount then it should automatically work 

totally makes sense
thanks for letting me know
